server/vertx-server/cats/src/main/scala/sttp/tapir/server/vertx/cats/VertxCatsServerInterpreter.scala

```diff
@@ -133,15 +133,17 @@ object VertxCatsServerInterpreter {
 
   implicit class VertxFutureToCatsF[A](f: => Future[A]) {
     def asF[F[_]: Async]: F[A] = {
-      Async[F].async_ { cb =>
-        f.onComplete({ handler =>
-          if (handler.succeeded()) {
-            cb(Right(handler.result()))
-          } else {
-            cb(Left(handler.cause()))
-          }
-        })
-        ()
+      Async[F].async { cb =>
+        Sync[F].delay {
+          f.onComplete({ handler =>
+            if (handler.succeeded()) {
+              cb(Right(handler.result()))
+            } else {
+              cb(Left(handler.cause()))
+            }
+          })
+          Some(().pure[F])
```

Comment:
isn't this fake cancellation logic - shouldn't we wait for `f` to complete?

Reply:
If we want to wait for `f` to complete then we should return `None` here, which basically means leaving `async_` as before. The difference is that:
- in older CE (<3.5) canceling such a fiber would not wait for the future and finish immediately (leaving a leaked future)
- in CE  3.5 this IO is uncancellable, leading to potential deadlocks

If there's no way to cancel underlying future, we need to choose between keeping the previous behavior (canceling the outer fiber and leaving a leaked future) and agreeing to the new uncancellable behavior. I chose the first solution to keep consistent with previous behavior.

Reply:
Ok, well I guess if we can't do anything, we can't :)